Samples/RedundantHubSample/RedundantHubSample/Program.cs

```diff
@@ -0,0 +1,152 @@
+            if (string.IsNullOrEmpty(notificationOutcome.NotificationId))
+            {
+                Console.WriteLine($"Fcm has no outcome due to it is only available for Standard SKU pricing tier.");
+                return null;
+            }
+
+            return await WaitForThePushStatusAsync(nhClient, notificationOutcome);
+        }
+
+        private static async Task<NotificationDetails> WaitForThePushStatusAsync(RedundantNotificationHubClient nhClient, NotificationOutcome notificationOutcome)
+        {
+            var notificationId = notificationOutcome.NotificationId;
+            var state = NotificationOutcomeState.Enqueued;
+            var count = 0;
+            NotificationDetails outcomeDetails = null;
+            while ((state == NotificationOutcomeState.Enqueued || state == NotificationOutcomeState.Processing) && ++count < 10)
+            {
+                try
+                {
+                    Console.WriteLine($"Status: {state}");
+                    outcomeDetails = await nhClient.GetNotificationOutcomeDetailsAsync(notificationId);
+                    state = outcomeDetails.State;
+                }
+                catch (MessagingEntityNotFoundException)
+                {
+                    // It's possible for the notification to not yet be enqueued, so we may have to swallow an exception
+                    // until it's ready to give us a new state.
+                }
+                Thread.Sleep(1000);
+            }
+            return outcomeDetails;
+        }
+
+        private static void PrintPushOutcome(NotificationDetails details, bool isPrimary)
+        {
+            if (details.FcmOutcomeCounts != null)
+            {
+                Console.WriteLine($"Notification outcome for {(isPrimary ? "Primary" : "Backup")}: " + string.Join(",", details.FcmOutcomeCounts.Select(kv => $"{kv.Key}:{kv.Value}")));
+            }
+            else
+            {
+                Console.WriteLine($"No outcomes for {(isPrimary ? "Primary" : "Backup")}");
+            }
+            Console.WriteLine($"{(isPrimary ? "Primary" : "Backup")} error details URL: {details.PnsErrorDetailsUri}");
+        }
+    }
+
+    public class RedundantNotificationHubClient
+    {
+        private readonly INotificationHubClient _primaryNotificationHubClient;
+        private readonly INotificationHubClient _backupNotificationHubClient;
+
+        public RedundantNotificationHubClient(string primaryConnectionString, string backupConnectionString, string hubName)
+        {
+            _primaryNotificationHubClient = NotificationHubClient.CreateClientFromConnectionString(primaryConnectionString, hubName);
+            _backupNotificationHubClient = NotificationHubClient.CreateClientFromConnectionString(backupConnectionString, hubName);
+        }
+
+        public bool UsePrimaryForSend { get; set; } = true;
```

Comment:
Let's rename this to `DefaultNamespace` and make it an enum to support GetInstallation and ListInstallations examples